sdk/core/Azure.Core/src/Shared/LruCache.cs

```diff
@@ -1,10 +1,12 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Curious: what motivates adding this type to shared source in Azure.Core?  Would other libraries want to use it, and for what scenarios?  We have a goal to reduce shared source, so if such a type is needed broadly by Azure SDK libraries, would it make sense to make it a public type instead?

Reply:
Right now, this class is only used by the two Schema Registry serializers, the one in `Microsoft.Azure.Data.SchemaRegistry.ApacheAvro` and this one. The reasoning behind adding this as a shared source in `Azure.Core` included it being a very general class (nothing Schema Registry related) and it potentially being useful for other SDKs if it's available.
I don't have any concerns about making this a public type, but it's not needed broadly by many libraries at this point. I don't have any specific scenarios in mind that other libraries may want to use it for, but it could be considered as a solution for any library wanting to add caching to their SDK.
I debated making this an internal type in `Azure.Data.SchemaRegistry` to share between the two, which is still on the table if the shared source doesn't seem like a great home.